### ISSUE TITLE
What is the status of the deep effect collection branch?

### ISSUE THREAD

**Issue by bdwain** [raisemarketplace/redux-loop#68]
I was following the [thread](https://github.com/reactjs/redux/issues/1528) that the deep effect collection branch was based on initially and I was interested to see what came of it. Are there any plans to do something like that in master?

**Comment by lukewestby** [comment]
I tried it out in this branch: https://github.com/raisemarketplace/redux-loop/tree/deep_effect_collection

It seems to work okay, but it's a pretty drastic departure from what redux-loop is intended to be so I'd prefer not to release it here and instead let someone else try to build something effective using that method. In particular, I really like how straightforward it is to test a reducer with redux-loop. Gathering effects in this way really changes the testing strategy.

**tl;dr** redux-loop will not be implementing deep effect collection and will continue to maintain parity with Elm, but this doesn't mean that deep effect collection is a bad idea and if you think it's worthwhile I hope you'll build something great!

**Comment by lukewestby** [comment]
I'm going to close this just so I can keep track of the open issues that represent bugs and things that need to be implemented, but we can definitely continue discussion if you'd like

**Comment by bdwain** [comment]
Cool. I was asking more out of curiosity. It sounds good in practice but I agree that having to wrap everything in a call to catcheffects or something like that will make testing more difficult. and testing in the return value is super easy.
